### PR TITLE
Get alias generator working

### LINKED ISSUE #5081: V2: alias generator
support both camel case and pascal case. Related to #3464.

Python function that modifies core schema.

### PR DESCRIPTION
**Pull request by dmontagu**
Closes #5081

Selected Reviewer: @samuelcolvin

### CODE REVIEW

**Comment by dmontagu** [comment]
please review

**Comment by dmontagu** [comment]
please review

File: tests/test_aliases.py [hunk 1]
```diff
@@ -112,6 +113,8 @@ class Model(BaseModel):
 
 @pytest.mark.xfail(reason='working on V2')
 def test_alias_child_precedence():
+    # TODO: Need to decide what we are doing with fields in model_config
+    #   If dropping completely, should we drop this test? If not, is there a migration path?
```
Comment (dmontagu) [inline tests/test_aliases.py:117]: Same thing here — should we just drop this test?

File: tests/test_aliases.py [hunk 2]
```diff
@@ -26,6 +26,8 @@ class MyModel(BaseModel):
 
 @pytest.mark.xfail(reason='working on V2')
 def test_alias_generator_with_field_schema():
+    # TODO: Are we dropping Config.fields?
```
[the remark this answers is not in the document]
Comment (dmontagu) [reply tests/test_aliases.py:29]: On further reflection, I'm going to change this comment to say "TODO: Remove this test", and remove all tests using Config.fields in a separate PR.

File: tests/test_aliases.py (hunk 1 above)
Comment (dmontagu) [reply tests/test_aliases.py:117]: Note to self: Need to update this comment

File: tests/test_aliases.py [hunk 3]
```diff
@@ -55,7 +56,7 @@ class MyModel(BaseModel):
             model_config = ConfigDict(alias_generator=return_bytes)
             bar: Any
 
-    assert str(e.value) == "Config.alias_generator must return str, not <class 'bytes'>"
+    assert str(e.value) == IsStr(regex="alias_generator <function .*> must return str, not <class 'bytes'>")
```
Comment (dmontagu) [inline tests/test_aliases.py:59]: I purposely changed this so you would see what alias generator function was misbehaving. (If you define multiple for different models, I don't think you'd be able to tell from the traceback where it came from, and I didn't see any reason not to include it here.)

**Review by samuelcolvin — APPROVED** [review]
LGTM.

@dmontagu happy for you to merge unless you have more to do.

I think in V2 it would be nice to have `camel_case` and `pascal_case` alias generators, built in, but obviously that's not required to get to an alpha.

Please update.

File: tests/test_aliases.py [hunk 4]
```diff
@@ -171,17 +192,16 @@ class Child(Parent):
...
-    # TODO: alias_priority has been removed from `Field`. Should we re-add it?
-    #   Is there something new that can be used to replicate this functionality?
-    #   See discussion in https://github.com/pydantic/pydantic/pull/5181/files#r1137618854
-    #   Either way, if we don't re-add it to `Field`, remember to update the migration guide
+    # TODO:
+    #  Alternative 1: we could drop alias_priority and tell people to manually override aliases in child classes
+    #  Alternative 2: we could add a new argument `override_with_alias_generator=True` equivalent to `alias_priority=1`
     class Parent(BaseModel):
+        w: bool = Field(..., alias='w_')
         x: bool = Field(..., alias='abc', alias_priority=1)
```
Comment (dmontagu) [inline tests/test_aliases.py:205]: This is the only TODO I have left here. @samuelcolvin do you have any suggestions for what to do here?

If no strong opinions, I'm inclined to just keep it as is for backwards compatibility and drop the TODO comment.

**Comment by dmontagu** [comment]
please review

**Comment by dmontagu** [comment]
> I think in V2 it would be nice to have `camel_case` and `pascal_case` alias generators, built in, but obviously that's not required to get to an alpha.

Where would you want to put these? I have [some functions along those lines](https://github.com/dmontagu/fastapi-utils/blob/master/fastapi_utils/camelcase.py) with [some tests](https://github.com/dmontagu/fastapi-utils/blob/master/tests/test_camelcase.py) I'd be happy to move into here. (And rename them at the same time..)

Or happy to take another existing implementation if you've got one handy. I agree this would be a good thing for us to provide out-of-the-box.

**Comment by dmontagu** [comment]
If I don't hear anything about this in the next couple hours I'll just merge it and leave the question of handling alias_priority for later, when we are going through TODOs prior to release

**Comment by dmontagu** [comment]
Actually, I'm going to just merge this now, but if you provide any feedback on this later I will look at it and open a new PR addressing it.

**Pull request merged**